### PR TITLE
Py3 step5

### PR DESCRIPTION
**Pull request by dodobas**

### CODE REVIEW

File: temba/utils/export.py [hunk 1]
```diff
@@ -201,6 +220,10 @@ def save_file(self):
         """
         gc.collect()  # force garbage collection
 
+        if six.PY3:  # pragma: no cover
```
Comment (dodobas) [inline temba/utils/export.py]: there is a problem when using `csv.writer` with `openpyxl` because latter is using zip internally which expect files to be opened in binary mode, but csv.writer on PY3 requires files to be opened in text mode. The workaround was to set `delete=False` on the initial file, close it and open it again in binary mode on PY3

Personally it's strange that we are using csv.writer to create .xlsx files which are then 'procesed' by openpyxl, maybe we should use openpyxl for all export file creation ? Maybe this was a performance/memory problem ?

Comment (nicpottier) [reply temba/utils/export.py]: Ya I think this stuff was almost certainly in response to a performance or memory problem, all of this was tweaked various times. Of course we don't know how those will act with new versions of the library, so I guess I'd bias towards trying this and then fixing it (again) if we run into problems.

File: temba/flows/models.py [hunk 2]
```diff
@@ -2514,21 +2514,38 @@ def get_step_type(dest, rulesets, actionsets):
...
-                    existing.uuid = str(uuid4())
-                    existing.save(update_fields=('flow', 'uuid'))
+                    ruleset.flow = None
+                    ruleset.uuid = str(uuid4())
+                    ruleset.save(update_fields=('flow', 'uuid'))
+                else:
+                    seen_existing_rulesets[uuid] = ruleset
+
+            # delete ruleset which are not seen
+            RuleSet.objects.filter(uuid__in=existing_rulesets_to_delete).delete()
```
[the remark this answers is not in the document]
Comment (dodobas) [reply temba/flows/models.py]: this was part of the previous PR (https://github.com/nyaruka/rapidpro/pull/1736/files#diff-ac952ec358ed4e019095b25fcac469adL2521)

[the remark this answers is not in the document]
Comment (dodobas) [reply temba/flows/models.py]: hmm, RuleSet objects were not deleted ... and I introduced new behavior which did not break the tests ... one quick-fix PR coming up ...

File: temba/api/v2/fields.py [hunk 3]
```diff
@@ -133,7 +133,7 @@ def many_init(cls, *args, **kwargs):
         """
         list_kwargs = {'child_relation': cls(*args, **kwargs)}
         for key in kwargs.keys():
-            if key in serializers.MANY_RELATION_KWARGS:
+            if key in relations.MANY_RELATION_KWARGS:
```
Comment (rowanseymour) [inline temba/api/v2/fields.py]: why is this needed if the DRF version hasn't changed?

File: temba/contacts/models.py [hunk 4]
```diff
@@ -1264,11 +1264,11 @@ def prepare_fields(cls, field_dict, import_params=None, user=None):
         active_scheme = [scheme[0] for scheme in ContactURN.SCHEME_CHOICES if scheme[0] != TEL_SCHEME]
 
         # remove any field that's not a reserved field or an explicitly included extra field
-        for key in field_dict.keys():
-            if (key not in Contact.ATTRIBUTE_AND_URN_IMPORT_HEADERS) and key not in extra_fields and key not in active_scheme:
-                del field_dict[key]
-
-        return field_dict
+        return {
```
Comment (rowanseymour) [inline temba/contacts/models.py]: if you can't fit a dict comprehension on a single line - probably clearer to use a regular for loop

File: temba/api/v2/fields.py (hunk 3 above)
Comment (dodobas) [reply temba/api/v2/fields.py]: well, we will need to bump the version of DRF, once we start moving to Django2.0, well probably it will be Django 2.2 which is the next LTS version

File: temba/contacts/models.py (hunk 4 above)
Comment (dodobas) [reply temba/contacts/models.py]: do we have a written/downlodable code style convention document, that has an example for this particular case ? :D

File: temba/api/v2/fields.py (hunk 3 above)
Comment (rowanseymour) [reply temba/api/v2/fields.py]: Sure sure - I wasn't sure if this was compatible with the current version we're using but makes sense after your explanation on the call

File: temba/contacts/models.py (hunk 4 above)
Comment (rowanseymour) [reply temba/contacts/models.py]: We try to adhere to the [Django code style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/) but we've never documented particular examples

File: temba/flows/tests.py [hunk 5]
```diff
@@ -7001,7 +7001,7 @@ def test_empty_interrupt_state(self):
         # disconnect action from interrupt state
         ruleset = flow.rule_sets.first()
         rules = ruleset.get_rules()
-        interrupt_rule = filter(lambda rule: isinstance(rule.test, InterruptTest), rules)[0]
+        interrupt_rule = list(filter(lambda rule: isinstance(rule.test, InterruptTest), rules))[0]
```
Comment (rowanseymour) [inline temba/flows/tests.py]: minor thing but `list + filter + lambda` might as well be a list comprehension
